src/fielddef.ts

```diff
@@ -485,9 +498,34 @@ export function resetTitleFormatter() {
 }
 
 export function title(fieldDef: FieldDefBase<string>, config: Config) {
+  if (isFieldDef(fieldDef)) {
```

Comment:
Why would FieldDefBase needs to be rechecked if it is a field def? This seems wrong.
Also, if this is not a fieldDef, the last line outside this `if` would be wrong too.

Reply:
The reason for this was because all the `is*FieldDef` functions require a `ChannelDef` as their input and as a result would not accept a `FieldDefBase`, therefore, I couldn't pass `fieldDef` to them without first running it through `isFieldDef`.

Reply:
Hmm still doesn’t sound like we are solving the problem at its cause here  — let’s discuss more in person

Reply:
(We better try to remove these two weird changes and solve the problem in other places with a less hacky solution)

Reply:
For example, why don'tee make is*FieldDef require FieldDefBase instead of ChannelDef